### PR TITLE
Updated alexa-sdk to 1.0.21

### PR DESCRIPTION
**Pull request by rk-7**
Please fill in this template.

- [x] Use a meaningful title for the pull request. Include the name of the package modified.
- [x] Test the change in your own code. (Compile and run.)
- [x] Follow the advice from the [readme](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#make-a-pull-request).
- [x] Avoid [common mistakes](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#common-mistakes).
- [x] Run `npm run lint package-name` (or `tsc` if no `tslint.json` is present).

Select one of these and delete the others:

If changing an existing definition:
- [x] Provide a URL to documentation or source code which provides context for the suggested changes: https://github.com/alexa/alexa-skills-kit-sdk-for-nodejs/tree/1.0.21
- [x] If you are making substantial changes, consider adding a `tslint.json` containing `{ "extends": "dtslint/dt.json" }`.

### CODE REVIEW

**Comment by rk-7** [comment]
@typescript-bot I have fixed the failures indicated in the Travis CI log. 

The "alexa-sdk" should really depend on i18next officially, which has typescript version 2.3. However, due to [aws-lambda-mock-context](https://github.com/moskalyk/typed-aws-lambda-mock-context), which depends on alexa-sdk has typescript version 2.2. which prevents me from doing it.
@moskalyk Please update the typescript version.

**Comment by rk-7** [comment]
to authors (@petebeegle @hoo29 @pascalwhoop @blforce). Could you review this PR?

**Comment by hoo29** [comment]
Why have you removed the types ConfirmationStatuses and DialogStates and added possible values as comments above the properties?

**Comment by rk-7** [comment]
@hoo29 I will revert it. I had this warning "Use an interface instead of a type literal", but I just checked Microsoft/dtslint#production which doesn't have that rule anymore.

**Comment by hoo29** [comment]
In addition to that, it would be better to define enums or types for all of the type information you have put in the comments.

**Comment by rk-7** [comment]
@hoo29 Thanks. I had all possible enum values as enums when I wrote it, but since 2.2 doesn't support string enums, we have types to take care of the enum values for now. :)
Please review.

File: types/alexa-sdk/index.d.ts
Comment (hoo29) [inline types/alexa-sdk/index.d.ts]: Type information is in the comments and not the type annotations.

File: types/alexa-sdk/index.d.ts
Comment (hoo29) [inline types/alexa-sdk/index.d.ts]: Type information is in the comments and not the type annotations.

File: types/alexa-sdk/index.d.ts
Comment (hoo29) [inline types/alexa-sdk/index.d.ts]: should this be 'image: Image'?

File: types/alexa-sdk/index.d.ts
Comment (hoo29) [inline types/alexa-sdk/index.d.ts]: The JSDoc say hintType is optional but the parameter does not have an optional qualifier.

File: types/alexa-sdk/index.d.ts
Comment (hoo29) [inline types/alexa-sdk/index.d.ts]: The JSDoc say metadata is optional but the parameter does not have an optional qualifier.

**Review by hoo29 — COMMENTED** [review]
minor type annotation fixes

**Comment by hoo29** [comment]
Apart from those small things, great contribution! I can't merge it as I'm not a maintainer but hopefully it will get put in soon.

File: types/alexa-sdk/index.d.ts
Comment (rk-7) [reply types/alexa-sdk/index.d.ts]: It keeps throwing "no-redundant-jsdoc-2  JSDoc tag '@type' is redundant in TypeScript code." if we put type annotations. I have updated it with a type number.

File: types/alexa-sdk/index.d.ts
Comment (rk-7) [reply types/alexa-sdk/index.d.ts]: It keeps throwing "no-redundant-jsdoc-2  JSDoc tag '@type' is redundant in TypeScript code." if we put type annotations. I have updated it with a type number.

File: types/alexa-sdk/index.d.ts
Comment (rk-7) [reply types/alexa-sdk/index.d.ts]: Updated.

File: types/alexa-sdk/index.d.ts
Comment (rk-7) [reply types/alexa-sdk/index.d.ts]: Fixed.

File: types/alexa-sdk/index.d.ts
Comment (rk-7) [reply types/alexa-sdk/index.d.ts]: Fixed.

**Comment by rk-7** [comment]
@hoo29 I have fixed those minor annotations issues. Thanks. I appreciate the great review as well :)
Just hoping it gets put sooner, as I share the pain of having the typings not updated for so long.
👍 / 👎 ?

**Comment by rk-7** [comment]
Authors (@petebeegle @hoo29 @pascalwhoop @blforce). Could you review this PR and **approve** it?

**Comment by rk-7** [comment]
@dt-bot Please take a look at this PR and take further actions as appropriate.

**Review by petebeegle — APPROVED** [review]

**Review by hoo29 — APPROVED** [review]

**Pull request merged**
